src/allmydata/client.py

```diff
@@ -582,6 +639,136 @@ def init_storage(self):
         for ic in self.introducer_clients:
             ic.publish("storage", ann, self._node_key)
 
+
+    @defer.inlineCallbacks
+    def init_storage_plugins(self):
+        """
+        Load, register, and announce any configured storage plugins.
+        """
+        storage_plugin_names = self._get_enabled_storage_plugin_names()
+        plugins = list(self._collect_storage_plugins(storage_plugin_names))
+        # TODO What if some names aren't found?
+        announceable_storage_servers = yield self._create_plugin_storage_servers(plugins)
```

Comment:
IMO it should be an error if a name was listed, but that plugin isn't found (if that's what you mean here?)